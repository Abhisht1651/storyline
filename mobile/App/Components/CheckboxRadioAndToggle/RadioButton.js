// @flow

import React, { useState } from 'react'
import { View, TouchableOpacity } from 'react-native'
import styles from '../Styles/CheckboxRadioAndToggleStyles/RadioButtonStyle'

type Props = {
  onPress: () => {}
}

const RadioButton = (props: Props) => {
  const { onPress } = props

  const [active, setActive] = useState(false)

  const onPressHandler = () => {
    setActive(!active)
    if (!active) onPress()
  }
  return (
    <TouchableOpacity style={styles.circle} onPress={onPressHandler}>
      {active && <View style={styles.checkedCircle} />}
    </TouchableOpacity>
  )
}

export default RadioButton
