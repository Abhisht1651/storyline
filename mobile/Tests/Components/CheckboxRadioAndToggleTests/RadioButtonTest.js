import { shallow } from 'enzyme'
import React from 'react'

// import { TouchableOpacity } from 'react-native'
import RadioButton from '../../../App/Components/CheckboxRadioAndToggle/RadioButton'

describe('RadioButton', () => {
  it('should match to snapshot', () => {
    const component = shallow(<RadioButton />)
    expect(component).toMatchSnapshot()
  })
})
