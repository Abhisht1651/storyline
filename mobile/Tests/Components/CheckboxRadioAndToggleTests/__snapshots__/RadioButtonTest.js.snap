// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should match to snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#8F8FAF",
      "borderRadius": 50,
      "borderWidth": 1,
      "height": 25,
      "justifyContent": "center",
      "width": 25,
    }
  }
/>
`;
